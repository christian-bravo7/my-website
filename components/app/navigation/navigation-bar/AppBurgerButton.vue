<template>
  <button
    class="w-12 h-12 | p-3 | rounded-full | bg-gray-100 | app-burger-button"
    :class="{ 'z-50': isActive }"
    @click="onClick"
    @keydown.enter="onClick"
  >
    <div
      class="flex flex-col items-end justify-center | relative | w-full h-full | app-burger-button__wrapper"
      :class="{ 'app-burger-button__wrapper--active': isActive }"
    >
      <span class="absolute | block | w-9/12 | bg-black" />
      <span class="absolute | block | w-full | bg-black" />
      <span class="absolute | block | w-7/12 | bg-black" />
    </div>
  </button>
</template>

<script lang="ts">
import { Component, Emit, Vue } from 'nuxt-property-decorator';

@Component
export default class AppBurgerButton extends Vue {
  isActive: boolean = false;

  @Emit('click')
  onClick (): void {
    this.isActive = !this.isActive;
  }
}
</script>

<style lang="scss" scoped>
.app-burger-button {
  &__wrapper {
    span {
      height: 2px;
      transition-duration: 300ms;
      transition-property: transform, opacity;
    }

    span:nth-child(1) {
      transform: translateY(-6px);
    }

    span:nth-child(2) {
      transform: translateY(0);
    }

    span:nth-child(3) {
      transform: translateY(6px);
    }

    &--active {
      span:nth-child(1) {
        @apply w-full;

        transform: rotate(45deg);
      }

      span:nth-child(2) {
        opacity: 0;
      }

      span:nth-child(3) {
        @apply w-full;

        transform: rotate(-45deg);
      }
    }
  }
}
</style>
