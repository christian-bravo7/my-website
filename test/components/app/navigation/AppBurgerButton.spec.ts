import { mount } from '@vue/test-utils';

import AppBurgerButton from '@/components/app/navigation/AppBurgerButton.vue';

describe('AppBurgerButton', () => {
  it('should render correclty', () => {
    const wrapper = mount(AppBurgerButton);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should add active class when isActive prop is provided', () => {
    const wrapper = mount(AppBurgerButton, {
      propsData: {
        isActive: true,
      },
    });

    const linesContainer = wrapper.find('.app-burger-button__wrapper');

    expect(linesContainer.classes()).toContain('app-burger-button__wrapper--active');
  });

  it('should emit onClick event when button is clicked', async () => {
    const wrapper = mount(AppBurgerButton, {
      propsData: {
        isActive: false,
      },
    });

    const button = wrapper.find('.app-burger-button');

    await button.trigger('click');

    expect(wrapper.emitted('onClick')).toBeTruthy();
  });

  it('should emit onClick event when button is focus and enter key is pressed', async () => {
    const wrapper = mount(AppBurgerButton, {
      propsData: {
        isActive: false,
      },
    });

    const button = wrapper.find('.app-burger-button');

    button.trigger('keydown.enter');

    expect(wrapper.emitted('onClick')).toBeTruthy();
  });
});
