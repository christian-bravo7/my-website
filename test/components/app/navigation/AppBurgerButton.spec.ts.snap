// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBurgerButton should render correclty 1`] = `
<button class="w-12 h-12 | p-3 | rounded-full | bg-gray-100 dark:bg-blue-900 | app-burger-button">
  <div class="flex flex-col items-end justify-center | relative | w-full h-full | app-burger-button__wrapper"><span class="absolute | block | w-9/12 | bg-black dark:bg-gray-100"></span> <span class="absolute | block | w-full | bg-black dark:bg-gray-100"></span> <span class="absolute | block | w-7/12 | bg-black dark:bg-gray-100"></span></div>
</button>
`;
